src/hardware/mixer.cpp

```diff
@@ -2188,17 +2197,15 @@ static std::optional<AudioFrame> parse_volume(const std::string &s)
 	auto parts = split(s, ':');
 	if (parts.size() == 1) {
 		if (const auto v = to_volume(parts[0]); v) {
-			AudioFrame volume = {*v, *v};
-			return volume;
+			return AudioFrame(*v, *v);
 		}
 	}
 	// stereo volume value
 	else if (parts.size() == 2) {
 		const auto l = to_volume(parts[0]);
 		const auto r = to_volume(parts[1]);
 		if (l && r) {
-			AudioFrame volume = {*l, *r};
-			return volume;
+			return AudioFrame(*l, *r);
 		}
 	}
 	return {};
```

Comment:
use `return nullptr;`?

Reply:
I don't think `nullptr` works with the `optional<AudioFrame>` return type - but we could use `optional`'s special `std::nullopt` notation.

The default `{}` syntax feels more general though, because it's used to return an empty optional value, for any `optional<T>`.  But If we change the interface to return a full blown `AudioFrame` object (without the optional wrapper), then this notation would still hold true: `return {};` hands back an default-init'd AudioFrame.

Reply:
Good discussion here about `return {};` versus `return std::nullopt`: https://stackoverflow.com/questions/62933403/why-do-we-need-stdnullopt

Reply:
ah yes, please disregard the comment; I swear I was looking at something that directly returned a `float *`.